src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/SasTokenControllerTest.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.hmcts.reform.blobrouter.controllers;
+
+import com.azure.storage.common.Utility;
+import com.azure.storage.common.implementation.StorageImplUtils;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit.jupiter.web.SpringJUnitWebConfig;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import uk.gov.hmcts.reform.blobrouter.exceptions.ServiceConfigNotFoundException;
+
+import java.time.OffsetDateTime;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringJUnitWebConfig
+@SpringBootTest
+@AutoConfigureMockMvc
+public class SasTokenControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    public void should_generate_sas_token_when_service_is_configured() throws Exception {
+        String sasTokenResponse = mockMvc
+            .perform(get("/token/bulkscan"))
+            .andExpect(status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+
+        final ObjectNode node = new ObjectMapper().readValue(sasTokenResponse, ObjectNode.class);
+        final String sasToken = node.get("sas_token").asText();
+        assertThat(sasToken).isNotBlank();
+
+        Map<String, String[]> queryParams = StorageImplUtils.parseQueryStringSplitValues(Utility.urlDecode(sasToken));
+
+        assertThat(queryParams.get("sig")).isNotNull();//this is a generated hash of the resource string
+        assertThat(queryParams.get("sv")).contains("2019-02-02");//azure api version is latest
+
+        OffsetDateTime expiresAt = OffsetDateTime.parse(queryParams.get("se")[0]); //expiry datetime for the signature
+        OffsetDateTime now = OffsetDateTime.now();
+        assertThat(expiresAt).isBetween(now, now.plusSeconds(300));
+        assertThat(expiresAt).isBefore(now.plusSeconds(301));
+
+        assertThat(queryParams.get("sp")).contains("wl");//access permissions(write-w,list-l)
```

Comment:
Same is done in unit test. Perhaps we could go with just "verification" query params parsed successfully and single assertion made for let's say "most significant" parameter?

Reply:
Updated to verify just the "sig" parameter